client/app/components/dynamic-parameters/DateRangeParameter.jsx

```diff
@@ -0,0 +1,115 @@
+import { includes, isArray, isString } from 'lodash';
+import { isDynamicDateRange, getDynamicDateRange } from '@/services/query';
+import { DateRangeInput } from '@/components/DateRangeInput';
+import { DateTimeRangeInput } from '@/components/DateTimeRangeInput';
+import DynamicButton from '@/components/dynamic-parameters/DynamicButton';
+
+import './DynamicParameters.less';
+
+const DYNAMIC_DATE_OPTIONS = [
+  { name: 'This week',
+    value: 'd_this_week',
+    label: () => getDynamicDateRange('d_this_week').value()[0].format('MMM D') + ' - ' +
+                 getDynamicDateRange('d_this_week').value()[1].format('MMM D') },
+  { name: 'This month', value: 'd_this_month', label: () => getDynamicDateRange('d_this_month').value()[0].format('MMMM') },
+  { name: 'This year', value: 'd_this_year', label: () => getDynamicDateRange('d_this_year').value()[0].format('YYYY') },
+  { name: 'Last week',
+    value: 'd_last_week',
+    label: () => getDynamicDateRange('d_last_week').value()[0].format('MMM D') + ' - ' +
+                 getDynamicDateRange('d_last_week').value()[1].format('MMM D') },
+  { name: 'Last month', value: 'd_last_month', label: () => getDynamicDateRange('d_last_month').value()[0].format('MMMM') },
+  { name: 'Last year', value: 'd_last_year', label: () => getDynamicDateRange('d_last_year').value()[0].format('YYYY') },
+  { name: 'Last 7 days',
+    value: 'd_last_7_days',
+    label: () => getDynamicDateRange('d_last_7_days').value()[0].format('MMM D') + ' - Today' },
+];
+
+const DYNAMIC_DATETIME_OPTIONS = [
+  { name: 'Today',
+    value: 'd_today',
+    label: () => getDynamicDateRange('d_today').value()[0].format('MMM D') },
+  { name: 'Yesterday',
+    value: 'd_yesterday',
+    label: () => getDynamicDateRange('d_yesterday').value()[0].format('MMM D') },
+  ...DYNAMIC_DATE_OPTIONS,
+];
+
+function isValidDateRangeValue(value) {
+  return isArray(value) && value.length === 2 && moment.isMoment(value[0]) && moment.isMoment(value[1]);
+}
+
+class DateRangeParameter extends React.Component {
+  static propTypes = {
+    type: PropTypes.string,
+    className: PropTypes.string,
+    value: PropTypes.any, // eslint-disable-line react/forbid-prop-types
+    parameter: PropTypes.any, // eslint-disable-line react/forbid-prop-types
+    onSelect: PropTypes.func,
+  };
+
+  static defaultProps = {
+    type: '',
+    className: '',
+    value: null,
+    parameter: null,
+    onSelect: () => {},
+  };
+
+  onDynamicValueSelect = (dynamicValue) => {
+    const { onSelect, parameter } = this.props;
+    if (dynamicValue === 'static') {
+      onSelect(parameter.getValue());
+    } else {
+      onSelect(dynamicValue.value);
+    }
+  };
+
+  render() {
+    const { type, value, parameter, onSelect, className } = this.props;
+    const isDateTimeRange = includes(type, 'datetime-range');
+    const hasDynamicValue = isDynamicDateRange(value);
+    const options = isDateTimeRange ? DYNAMIC_DATETIME_OPTIONS : DYNAMIC_DATE_OPTIONS;
```

Comment:
Is there a difference between "Today / Yesterday" in Date and that in DateRange? (in terms of sql query perhaps?)

Reply:
Yes, for DateRanges you'll have two values so in SQL you'd have:

```sql
WHERE column BETWEEN '{{range.start}}' AND '{{range.end}}'
```

That's why I haven't enabled Today/Yesterday for Date Ranges without time, it seemed to have no sense as you wouldn't have the "Range".